### ISSUE TITLE
OME-M 1.40.10 breaks ome_device_power_settings

### ISSUE THREAD

**Issue by MallocArray** [dell/dellemc-openmanage-ansible-modules#353]
##### Summary

Existing playbook using ome_device_power_settings to configure MX7000 chassis power settings works fine on 1.30.00 until upgraded to 1.40.10 when the same playbook fails with "msg": "'PowerBudgetOverride'"

Collection version 4.4.40

##### Component Name

ome_device_power_settings

##### Ansible Version

```paste below
ansible 2.10.16.post0
  config file = /runner/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.8 (default, Aug 25 2021, 16:13:02) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
```

##### iDRAC or OpenManage Enterprise version

OME-M 1.40.10

##### Steps to Reproduce

Existing playbook using ome_device_power_settings to configure MX7000 chassis power settings works fine on 1.30.00 until upgraded to 1.40.10 when the same playbook fails with "msg": "'PowerBudgetOverride'"

##### Playbook used

```paste below
    - name: OME-M Chassis Power Settings - Per Chassis
vars:
  chassis_group_members:
    - name: DellChassis01
      service_tag: xxxxx
      primary_grid: GRID_1
      data_center: Foo
      room: Datacenter
      aisle: 6
      rack: bar
      rack_slot: 38
      location: bar

      dellemc.openmanage.ome_device_power_settings:
        hostname: "{{ inventory_hostname }}"
        username: "{{ dell_ome_username }}"
        password: "{{ dell_ome_password }}"
        device_service_tag: "{{ item.service_tag }}"
        redundancy_configuration:
          redundancy_policy: "{{ item.redundancy_policy | default('GRID_REDUNDANCY') }}"
        hot_spare_configuration:
          enable_hot_spare: "{{ item.enable_hot_spare | default('true') }}"
          primary_grid: "{{ item.primary_grid | default('GRID_1') }}"
      loop: "{{ chassis_group_members }}"
      loop_control:
        label: "{{ item.name }}"
      tags:
        - power

```

##### Expected Results

As the settings are already configured, expect to get a return of OK with no changes

##### Actual Results

The full traceback is:
  File "/tmp/ansible_dellemc.openmanage.ome_device_power_settings_payload_hm5c6003/ansible_dellemc.openmanage.ome_device_power_settings_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/ome_device_power_settings.py", line 331, in main
  File "/tmp/ansible_dellemc.openmanage.ome_device_power_settings_payload_hm5c6003/ansible_dellemc.openmanage.ome_device_power_settings_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/ome_device_power_settings.py", line 294, in fetch_device_details
  File "/tmp/ansible_dellemc.openmanage.ome_device_power_settings_payload_hm5c6003/ansible_dellemc.openmanage.ome_device_power_settings_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/ome_device_power_settings.py", line 232, in check_mode_validation
failed: [dellgroup.domain.com] (item=DellChassis01) => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/libexec/platform-python"
    },
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "device_id": null,
            "device_service_tag": "xxxxx",
            "hostname": "dellgroup.domain.com",
            "hot_spare_configuration": {
                "enable_hot_spare": true,
                "primary_grid": "GRID_1"
            },
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "power_configuration": null,
            "redundancy_configuration": {
                "redundancy_policy": "GRID_REDUNDANCY"
            },
            "username": "redacted@domain.com"
        }
    },
    "item": {
        "aisle": 6,
        "data_center": "foo",
        "location": "bar",
        "name": "DellChassis01",
        "primary_grid": "GRID_1",
        "rack": "bar",
        "rack_slot": 38,
        "room": "Datacenter",
        "service_tag": "xxxxx"
    },
    "msg": "'PowerBudgetOverride'"
}

#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by anupamaloke** [comment]
@MallocArray, this will be fixed with v5.0.0 that will be released in a couple of weeks down the line.

**Comment by MallocArray** [comment]
@anupamaloke  If there is a known fix coming, could the changed lines be made available or point to a commit that will resolve it?  Since I've already upgraded the firmware, my playbooks will continue to fail for this chassis group unless I disable the task entirely.  If I could change my local copy of the module to have the fix, I could limp along until the 5.0.0 release officially comes out

**Comment by anupamaloke** [comment]
@MallocArray, let me check internally to find out if we can share the code-snippets.

**Comment by sachin-apa** [comment]
@MallocArray  Please make the changes in a module as below.
https://github.com/dell/dellemc-openmanage-ansible-modules/blob/e10c98ed6a414d15a0920fa2c51c3760d3efc5ec/plugins/modules/ome_device_power_settings.py#L228

Assign the _power_data_ variable as below
` power_data = {"PowerCap": loc_data.get("PowerCap"), "MinPowerCap": loc_data["MinPowerCap"],
                  "MaxPowerCap": loc_data["MaxPowerCap"], "RedundancyPolicy": loc_data.get("RedundancyPolicy"),
                  "EnablePowerCapSettings": loc_data["EnablePowerCapSettings"],
                  "EnableHotSpare": loc_data["EnableHotSpare"], "PrimaryGrid": loc_data.get("PrimaryGrid")}`

**Comment by anupamaloke** [comment]
@MallocArray, this isssue is fixed as part of #355. Please reopen if you still run into this issue.
